### PR TITLE
 updated OneLogin to v3p 7.2 compliant

### PR DESCRIPTION
**Pull request by Alastairkn**
updated OneLogin to v3 which has removed mcrypt which will make this module php 7.2 compliant

### CODE REVIEW

File: src/Control/SAMLController.php [hunk 1]
```diff
@@ -52,19 +51,19 @@ public function index()
...
      * LDAP side of this module to finish off loading Member data.
      *
-     * @throws OneLogin_Saml2_Error
+     * @throws OneLogin\Saml2\Error
      * @throws \Psr\Container\NotFoundExceptionInterface
      */
     public function acs()
     {
-        /** @var \OneLogin_Saml2_Auth $auth */
+        /** @var \OneLogin\Saml2\Auth $auth */
```
Comment (NightJar) [inline src/Control/SAMLController.php]: `Auth` is sufficient here, as opposed to the FQCN.

File: src/Control/SAMLController.php [hunk 2]
```diff
@@ -183,7 +182,7 @@ public function acs()
     public function metadata()
     {
         try {
-            /** @var OneLogin_Saml2_Auth $auth */
+            /** @var OneLogin\Saml2\Auth $auth */
```
Comment (NightJar) [inline src/Control/SAMLController.php]: As earlier, the namespace is unnecessary, as the class is imported above (`use ...`)

File: src/Control/SAMLController.php [hunk 3]
```diff
@@ -192,9 +191,9 @@ public function metadata()
                 header('Content-Type: text/xml');
                 echo $metadata;
             } else {
-                throw new \OneLogin_Saml2_Error(
+                throw new \OneLogin\Saml2\Error(
```
Comment (NightJar) [inline src/Control/SAMLController.php]: Please import the class (`use OneLogin\Saml2\Error;`) and use the unqualified classname here.

File: src/Control/SAMLController.php [hunk 4]
```diff
@@ -192,9 +191,9 @@ public function metadata()
...
                     'Invalid SP metadata: ' . implode(', ', $errors),
-                    \OneLogin_Saml2_Error::METADATA_SP_INVALID
+                    \OneLogin\Saml2\Error::METADATA_SP_INVALID
```
Comment (NightJar) [inline src/Control/SAMLController.php]: and here - unqualify the classname via use statements please

File: src/Helpers/SAMLHelper.php [hunk 5]
```diff
@@ -38,12 +38,12 @@ class SAMLHelper
     public $SAMLConfService;
 
     /**
-     * @return OneLogin_Saml2_Auth
+     * @return OneLogin\Saml2\Auth
```
Comment (NightJar) [inline src/Helpers/SAMLHelper.php]:
```diff
- OneLogin\Saml2\Auth
+ Auth
```

File: src/Control/SAMLController.php [hunk 6]
```diff
@@ -52,19 +51,19 @@ public function index()
      * if not existent), with the user already logged in. Login triggers memberLoggedIn hooks, which allows
      * LDAP side of this module to finish off loading Member data.
      *
-     * @throws OneLogin_Saml2_Error
+     * @throws OneLogin\Saml2\Error
```
Comment (NightJar) [inline src/Control/SAMLController.php]: please just use `Error` once you've added the `use` statement (see other comments)

**Review by NightJar — APPROVED** [review]
Looks really good, just a few minor formatting tidies please :)
This is really awesome, thanks for contributing!

**Review by NightJar — CHANGES_REQUESTED** [review]
whoops, accidentally approved instead of requesting changes

**Comment by Alastairkn** [comment]
No worries, i'll make the changes you've said and make a new pull request

I did wonder about the fqcn.

**Comment by madmatt** [comment]
Oooh yay thanks for this @Alastairkn, I was wanting to do this as well but when we used the module they hadn't released v3 yet stable and the testing branch for it was a bit broken.

+1 for merging this one in as soon as the changes have been made as @NightJar requested, and then tagging `1.0.0` to get a stable version released IMO.

**Review by NightJar — APPROVED** [review]
Most excellent, thanks for this!

**Comment by NightJar** [comment]
So tag `1.0.0` your reckon @madmatt ?
I can do that :D

**Comment by NightJar** [comment]
Oooh, this also closes #12 - yay \o/

**Comment by NightJar** [comment]
@Alastairkn @madmatt I've tagged `1.0.0` :)

@madmatt I don't have context over what is included in this, so if you have a spare moment it'd be awesome if you could [add some release notes](https://github.com/silverstripe/silverstripe-saml/releases/edit/1.0.0) 👍 
I know you're busy though, so no big stress :)

**Comment by madmatt** [comment]
Perfect, thanks @NightJar. I've updated the release notes now :)

**Pull request merged**